src/vmm/src/controller.rs

```diff
@@ -57,19 +57,6 @@ impl VmmController {
             .started
     }
 
-    /// Sets a vsock device to be attached when the VM starts.
-    pub fn set_vsock_device(&mut self, config: VsockDeviceConfig) -> UserResult {
-        if self.is_instance_initialized() {
```

Comment:
I don't think it is OK to delete this check. This function is also called as an external API call and can be called anytime.

Reply:
Please note you haven't made vsock config as OperationNotSupportedPostBoot.

Reply:
Good catch!

Reply:
Actually, I've checked and all external API calls are covered in both pre-boot and post-boot handlers. They actually **have to** be covered as a `match` in rust doesn't allow incomplete coverage of a matched value's range.